Valentyn Genkin APIs week1

ValentynGenkin opened the pull request.

allhandsondeck commented:
I think we should only include the Using APIs homework files in this pull request.
Could you please make sure excluding the rest?

allhandsondeck commented on 3-UsingAPIs/Week1/homework/ex1-johnWho.js:
```diff
@@ -8,18 +12,17 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
```
Do we still need setTimeout?

allhandsondeck commented on 3-UsingAPIs/Week1/homework/ex1-johnWho.js:
```diff
@@ -1,4 +1,8 @@
 'use strict';
+
+// eslint-disable-next-line no-unused-vars
+const { reject } = require('lodash');
```
I think we don't use this.
Can we delete it?

allhandsondeck commented on 3-UsingAPIs/Week1/homework/ex3-rollDie.js:
I think we are missing this requirement of the homework
> Does the problem described above still occur? If not, what would be your explanation? Add your answer as a comment to be bottom of the file.

allhandsondeck commented on 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:
I think we are missing this requirement of the homework
> You may also notice that, in the case of a rejected promise, dice that have not yet finished their roll continue to do so. Can you explain why? Please add your answer as a comment to the end of the exercise file.

allhandsondeck commented on 3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js:
```diff
@@ -17,12 +17,23 @@ const rollDie = require('../../helpers/pokerDiceRoller');
   const results = [];
 
-  // TODO: expand the chain to include five dice
   return rollDie(1)
```
Good solution 👍🏼

allhandsondeck reviewed (changes requested):
Nice job Valentyn!

Please make sure resolving my comments by clicking the `resolve conversation` button after pushing your changes.

ValentynGenkin commented:
Hi Utku. Thanks for you feedback. I added comment and deleted enlist comments // eslint-disable-next-line no-unused-vars.
But I stuck with remove previous commits, could you help me with this?

allhandsondeck reviewed (approved):
Thanks for your changes Valentyn.
Please ignore my comment about excluding the previous homework files. It would suffice in the next homework having only the related homework files in the PR.
